### PR TITLE
jsbn: change "declare namespace" to "declare module"

### PR DESCRIPTION
**Pull request by altabayoyon**
Please fill in this template.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x] Test the change in your own code. (Compile and run.)
- [x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [see explanation below ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Issue:
    Error in @types/jsbn Typescript declarations file.

Symptom:
    After installing typescript, jsbn and @types/jsbn in a project,
    creation of a Typescript file that import of BigInteger with
        import {BigInteger} from 'jsbn';
    results in
        TS2306: File <path-to>node_modules/@types/jsbn/index.d.ts is not a module.

Resolution:
    In @types/jsbn/index.d.ts, replace line
        declare namespace jsbn {
    with
        declare module 'jsbn' {

Additional Notes:
    When using current @types/jsbn as is,
    using Webstorm's "quickfix"  to resolve a reference to BigInteger
       example: const bi = new BigInteger('2', 16);
    may result in adding
        import BigInteger = jsbn.BigInteger;
    which will work with local builds but if it becomes a published npm
    package (lets call it zzz), a browser-side application (e.g. Angular)
    importing package zzz will result in a JavaScript runtime error
    as it is unable to resolve 'jsbn.BigInteger'.

### CODE REVIEW

**Review by Evgenus — APPROVED** [review]

File: types/jsbn/index.d.ts
Comment (rbuckton) [inline types/jsbn/index.d.ts]: It looks like jsbn supports CommonJS (when it detects an `exports` object) and a global namespace (when there is no `exports` object).

A better approach would be to remove the `declare module 'jsbn' {` entriely (making this file into an external module), and adding an `export as namespace jsbn;` line to the file. That will allow it to work in both CommonJS and globally.

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/jsbn/index.d.ts
Comment (rbuckton) [reply types/jsbn/index.d.ts]: If you do this, you should also export the `RandomGenerator` and `Reduction` interfaces

**Comment by RyanCavanaugh** [comment]
@al2xed Please address comments from the code reviewers.

File: types/jsbn/index.d.ts
Comment (al2xed) [reply types/jsbn/index.d.ts]: Will give it a try then test.  Schedule currently full ... please be patient.

**Comment by al2xed** [comment]
1. Made revisions as suggested by rbuckton (thank you!)
    + Removed the **declare module 'jsbn' {**
    + Added **export as namespace jsbn;**
    + export RandomGenerator and Reduction interface
2. Tested in a library package ... success!
3. Tested in an application that installs the library package ... success!
4. Committed and pushed to **master** branch on **al2xed/DefinitelyTyped**

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Comment by RyanCavanaugh** [comment]
@rbuckton - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Review by rbuckton — APPROVED** [review]

**Pull request merged**
